### PR TITLE
Fix upper cases in media expressions, fix test failures related to platform-dependent error output with glyphs, fix failure to comma-separate keyframes percentage list

### PR DESCRIPTION
**Pull request by zzwx**
Attempt 2, taking into account that some platforms may have unicode glyphs to describe the source spans, and some may not.

### CODE REVIEW

**Comment by zzwx** [comment]
Trying to fix dart-lang/tools#1172 as well. I guess I'm not correctly creating pull requests

**Comment by zzwx** [comment]
I've tested it on both Windows and Linux, tests don't fail on both. I'm hoping you'll give it a green light to run the checks here.

**Comment by natebosch** [comment]
Thanks for the PR!

We prefer to keep refactoring and other fixes in separate PRs from the main fix. I also think we probably want to disable ascii glyphs for all the tests instead of patching the expectations. I'll start looking at that in https://github.com/dart-lang/csslib/pull/142

**Comment by zzwx** [comment]
> Thanks for the PR!
> 
> We prefer to keep refactoring and other fixes in separate PRs from the main fix. I also think we probably want to disable ascii glyphs for all the tests instead of patching the expectations. I'll start looking at that in dart-lang/csslib#142

Sure, as long as it fixes the bugs!

**Comment by natebosch** [comment]
@zzwx - I don't have rights to push to your fork (when you open a PR with the same account that owns the repository, it by default allows pushes from the repo maintainers, but using a separate account for the repo may have interefered with that).

https://github.com/dart-lang/csslib/commit/4140cca8c10946b1c3ef26a52069a61cb9b0c51a has the merge if you want to bring it to your PR branch.

File: pubspec.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: csslib
-version: 0.17.1
+version: 0.17.3
```
Comment (natebosch) [inline pubspec.yaml]:
```suggestion
version: 0.17.2
```

Version `0.17.2` hasn't been published yet.

**Review by natebosch — APPROVED** [review]

File: pubspec.yaml (hunk 1 above)
Comment (zzwx) [reply pubspec.yaml]: Oops sorry! Right

**Comment by zzwx** [comment]
Will this allow you to resolve the conflicts? What a mess I have made :)

**Comment by natebosch** [comment]
> Will this allow you to resolve the conflicts? What a mess I have made :)

If it's feeling like a mess I'm happy to take over 😄 

I can start this in a new PR based on this one so that I can merge with the conflicts resolved.

**Comment by natebosch** [comment]
Follow along in https://github.com/dart-lang/csslib/pull/143

**Pull request closed without merging**